### DISCUSSION TITLE
Integrating Amber API data with EMHASS + Solaredge Battery

### DISCUSSION THREAD

**Discussion by szadro** [amberelectric/public-api#142] [Home Assistant]
I am working to integrate Amber API data with EMPHASS to better manage my deferable loads and EV charger.

I have hit snag controlling my Solaredge System/battery via Home-Assistant-Solaredge-Modbus.  My system continues to revert back to Maximize Self Consumptions and after I attempt to take control.  For example, when I set "Storage Control Model" to "Remote Control", then inverter behaves as expected.  However, after 5-10 seconds "Storage Control Mode" reverts back to "Maximize Self Consumption" and the inverter stops charging from the grid (See the attached screenshot). I have played around with the Storage Remote Control Timeout setting which has no impact.

Has anyone experienced this issue / can suggest a fix?

**Comment by szadro** [comment]
BTW my EV charging is really messing with SmartShift hence the desire to move to EMHASS

**Reply by purcell-lab** [reply]
I had exactly this issue.

The problem is SmartShift is still trying to control your inverter. I had to debug with SolarEdge support and eventually had to get my inverter unenrolled by Amber. Just turning off SmartShift in the app isn't sufficient.

I also have issue with my EV charging and SmartShift conflicting, I have similar conflicts with SmartShift and ChargeHQ. This issue is because the systems aren't aware of each other.

Using EMHASS reduces this issue and when it schedules charging my EV I also have EMHASS set the battery backup reserve so it doesn't discharge my home battery when charging the EV, unless that is actually what EMHASS wants.

**Comment by szadro** [comment]
Thanks for sharing your experience and tips.
